docs/DesignDocs/SGX_QuoteEx_Integration.md

```diff
@@ -169,18 +169,22 @@ by working with an AESM service on the local platform.
 
 ### Priority between the SGX DCAP and quote-ex Libraries
 
-An SGX platform can have either the DCAP library or the quote-ex library,
-or both of them installed.
-
-* When only the DCAP library is installed, it will be used.
-* When only the quote-ex library is installed, it will be used.
-* When both libraries are installed:
-    * If the DCAP library is configured for in-process quote generation, the DCAP
-    library is chosen, so this default behavior for existing applications is
-    preserved.
-    * Otherwise if the DCAP library is configured for out-of-process quote
-    generation, the quote-ex library will be used, as it supports more evidence
-    formats.
+An SGX platform can have either the DCAP library or the quote-ex library, or both of them installed.
+
+* If `SGX_AESM_ADDR` is not set, then the DCAP library must be installed. During quote generation, SGX quoting enclaves will be loaded in the application process.
+* If `SGX_AESM_ADDR` is set, then the quote-ex library as well as the AESM plugins must be installed. During quote generation, SGX quoting enclaves will be loaded in the SGX AESM service running in kernel mode. The necessary libraries and plugins are listed as follows:
+  * `libsgx-quote-ex`
+  * `sgx-aesm-service`
+  * `libsgx-aesm-ecdsa-plugin`
+  * `libsgx-aesm-pce-plugin`
+  * `libsgx-aesm-quote-ex-plugin`
+
+On Linux platforms running kernel 5.11 or later, to be able to use the DCAP library for in-process quote generation, the user running the process needs to be added to the `sgx_prv` group, with the following command:
+* `sudo usermod -a -G sgx_prv <username>`
+
+Note - if this group has not been created, create it before adding the user to the group.
+
+For more information, see [SGX DCAP in-proc quote](https://github.com/intel/SGXDataCenterAttestationPrimitives/blob/master/driver/linux/README.kernel.md#apps-with-in-process-quote).
```

Comment:
The referenced doc does not explain how to create the "sgx_prv" group, if SGX DCAP driver BIN file is not used, for example, in a Linux distro with kernel 5.11.

Reply:
@haitaohuang, do you have a suggestion on the instruction to create the "sgx_prv" group, if SGX DCAP driver is not used?

Reply:
Currently (2.13-2.10) sgx-aesm-service create the group. We are moving that to libsgx-ae-pce in next release. see https://github.com/intel/linux-sgx/issues/678

So in future, regardless in-proc or out-of-proc, you just need install libsgx-ae-pce. Otherwise, to manually create user group: 
```
# groupadd sgx_prv
```

Create  and trigger udev rules:
```
# cat > /etc/udev/rules.d/90-sgx-v40.rules <<EOF
SUBSYSTEM=="misc",KERNEL=="sgx_enclave",MODE="0666",SYMLINK+="sgx/enclave"
SUBSYSTEM=="misc",KERNEL=="sgx_provision",GROUP="sgx_prv",MODE="0660",SYMLINK+="sgx/provision"
EOF
# udevadm trigger

```

Reply:
Is it necessary to create and trigger the userspace dev rules before using in-proc quote generation? It seems that, without doing the second part, simply doing the first part works for me.

Reply:
On a Linux system with kernel lower than 5.11 and DCAP driver lower than1.41, sgx_prv is not required for in-proc attestation. @ryanhsu19 , what's the system configuration (SGX driver) when you did the experiment?

Reply:
I opened a new VM with Ubuntu 18.04 LTS and a v5.40 kernel. Then I installed `sgx_linux_x64_driver_1.41.bin` manually.

Reply:
if you use the bin installer, it will do those steps for you.

Reply:
@haitaohuang, will https://github.com/intel/SGXDataCenterAttestationPrimitives/blob/master/driver/linux/README.kernel.md#apps-with-in-process-quote be updated to include the information you posted in this thread?